sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1069,11 +1110,19 @@ class Analyzer(
         case a @ Aggregate(grouping, expressions, child) =>
           failOnOuterReference(a)
           val referencesToAdd = missingReferences(a)
+          if (foundNonEqualCorrelatedPred) {
```

Comment:
Move the `if (foundNonEqualCorrelatedPred)` into the `failOnNonEqualCorrelatedPredicate` function